src/commonMain/kotlin/org/cru/godtools/tool/state/State.kt

```diff
@@ -0,0 +1,23 @@
+package org.cru.godtools.tool.state
+
+import org.cru.godtools.tool.internal.parcel.Parcelable
+import org.cru.godtools.tool.internal.parcel.Parcelize
```

Comment:
So these are coming from the same file ```src/commonMain/kotlin``` folder, so does each folder/platform reference its own ```Parcelable```? For example if I wanted to make a ```State.kt``` inside of ```src/jsMain``` it would be referencing the ```Parcelable``` from that directory?

Reply:
yes, when KMP builds a target platform it combines the source from `commonMain` and the source directory for the target platform.

those 2 imports are defined as `expect` within `commonMain`, then each target platform source directory provides its own `actual` implementation of them.

So when this logic is compiled for iOS, it uses the iOS `actual` implementation vs android where it would use the android `actual` implementation.

Reply:
so, creating a `State` object in `jsMain` would actually throw a compile error because that `State` object would conflict with the `State` object defined in `commonMain`.

In addition, logic in `commonMain` can't directly reference logic out of `jsMain` because that same logic isn't available for `iosMain`. The `expect`/`actual` concept works around this by defining in `commonMain` what is "expected" to be provided by the "actual" platform source set